src/pages/iou/request/step/IOURequestStepUpgrade.tsx

```diff
@@ -1,36 +1,113 @@
 import UpgradeConfirmation from '@pages/workspace/upgrade/UpgradeConfirmation';
 import UpgradeIntro from '@pages/workspace/upgrade/UpgradeIntro';
-import {setMoneyRequestParticipants} from '@userActions/IOU';
+import {setCustomUnitRateID, setMoneyRequestParticipants} from '@userActions/IOU';
 import CONST from '@src/CONST';
 import * as Policy from '@src/libs/actions/Policy/Policy';
+import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import type SCREENS from '@src/SCREENS';
 
+const DEFAULT_FEATURE_NAME = 'categories';
+
 type IOURequestStepUpgradeProps = PlatformStackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.STEP_UPGRADE>;
 
 function IOURequestStepUpgrade({
     route: {
-        params: {transactionID, action},
+        params: {transactionID, action, featureName = DEFAULT_FEATURE_NAME},
     },
 }: IOURequestStepUpgradeProps) {
     const styles = useThemeStyles();
-    const feature = CONST.UPGRADE_FEATURE_INTRO_MAPPING.categories;
+    const feature = useMemo(
+        () =>
+            Object.values(CONST.UPGRADE_FEATURE_INTRO_MAPPING)
+                .filter((value) => value.id !== CONST.UPGRADE_FEATURE_INTRO_MAPPING.policyPreventMemberChangingTitle.id)
+                .find((f) => f.alias === featureName),
+        [featureName],
+    );
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+    const personalDetails = usePersonalDetails();
+
+    const [transaction] = useOnyx(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {canBeMissing: true});
 
     const [isUpgraded, setIsUpgraded] = useState(false);
     const policyDataRef = useRef<CreateWorkspaceParams | null>(null);
+    const isDistanceRateUpgrade = featureName === CONST.UPGRADE_FEATURE_INTRO_MAPPING.distanceRates.alias;
+    const isCategorizing = featureName === CONST.UPGRADE_FEATURE_INTRO_MAPPING.categories.alias;
+
+    const onConfirmUpgrade = useCallback(() => {
+        const reportID = policyDataRef.current?.expenseChatReportID;
+        const policyID = policyDataRef.current?.policyID;
+        setMoneyRequestParticipants(transactionID, [
+            {
+                selected: true,
+                accountID: 0,
+                isPolicyExpenseChat: true,
+                reportID,
+                policyID,
+                searchText: policyDataRef.current?.policyName,
+            },
+        ]);
+        Navigation.goBack();
+
+        switch (feature?.id) {
+            case CONST.UPGRADE_FEATURE_INTRO_MAPPING.distanceRates.id: {
+                if (!policyID || !reportID) {
+                    return;
+                }
+                setTransactionReport(transactionID, {reportID}, true);
+                // Let the confirmation step decide the distance rate because policy data is not fully available at this step
+                setCustomUnitRateID(transactionID, '-1');
```

Comment:
Can we set this to undefined instead? We shouldn't default strings to any value and this kind of violates that principle.